sdk/azcore/runtime/policy_retry.go

```diff
@@ -59,13 +59,33 @@ func setDefaults(o *policy.RetryOptions) {
 }
 
 func calcDelay(o policy.RetryOptions, try int32) time.Duration { // try is >=1; never 0
-	delay := time.Duration((1<<try)-1) * o.RetryDelay
+	// avoid overflow when shifting left
+	factor := time.Duration(math.MaxInt64)
+	if try < 63 {
+		factor = time.Duration(int64(1<<try) - 1)
+	}
+
+	delay := factor * o.RetryDelay
+	if delay < factor {
+		// overflow has happened so set to max value
+		delay = time.Duration(math.MaxInt64)
+	}
+
+	// Introduce jitter:  [0.0, 1.0) / 2 = [0.0, 0.5) + 0.8 = [0.8, 1.3)
+	jitterMultiplier := rand.Float64()/2 + 0.8 // NOTE: We want math/rand; not crypto/rand
+
+	delayFloat := float64(delay) * jitterMultiplier
+	if delayFloat > float64(math.MaxInt64) {
+		// the jitter pushed us over MaxInt64, so just use MaxInt64
+		delay = time.Duration(math.MaxInt64)
+	} else {
+		delay = time.Duration(delayFloat)
+	}
 
-	// Introduce some jitter:  [0.0, 1.0) / 2 = [0.0, 0.5) + 0.8 = [0.8, 1.3)
-	delay = time.Duration(delay.Seconds() * (rand.Float64()/2 + 0.8) * float64(time.Second)) // NOTE: We want math/rand; not crypto/rand
 	if delay > o.MaxRetryDelay {
```

Comment:
Apparently we've had a bug in here for awhile:
```suggestion
	if o.MaxRetryDelay > 0 && delay > o.MaxRetryDelay {
```

Reply:
I was expecting the MaxRetryDelay being back filled at here:

https://github.com/Azure/azure-sdk-for-go/blob/1fbf735512c6be7673c97c9f2843a27aede2dc42/sdk/azcore/runtime/policy_retry.go#L37-L42

Reply:
Correct, `o.MaxRetryDelay` will never be zero or negative here (I had the same thought then realized this after making the same fix).

Reply:
We should add a comment here explaining this.

Reply:
I updated with a comment, PTAL